tests/link.js

```diff
@@ -0,0 +1,251 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals document */
+
+import ClassicTestEditor from '/tests/core/_utils/classictesteditor.js';
+import testUtils from '/tests/core/_utils/utils.js';
+import { keyCodes } from '/ckeditor5/utils/keyboard.js';
+import { setData as setModelData } from '/tests/engine/_utils/model.js';
+
+import Link from '/ckeditor5/link/link.js';
+import LinkEngine from '/ckeditor5/link/linkengine.js';
+import Button from '/ckeditor5/ui/button/button.js';
+import LinkBalloonPanel from '/ckeditor5/link/ui/linkballoonpanel.js';
+
+import ClickObserver from '/ckeditor5/engine/view/observer/clickobserver.js';
+
+testUtils.createSinonSandbox();
+
+describe( 'Link', () => {
+	let editor, linkFeature, linkButton, unlinkButton, balloonPanel, editorElement;
+
+	beforeEach( () => {
+		editorElement = document.createElement( 'div' );
+		document.body.appendChild( editorElement );
+
+		return ClassicTestEditor.create( editorElement, {
+			features: [ Link ]
+		} )
+			.then( newEditor => {
+				newEditor.editing.view.attachDomRoot( editorElement );
+
+				editor = newEditor;
+
+				linkFeature = editor.plugins.get( Link );
+				linkButton = editor.ui.featureComponents.create( 'link' );
+				unlinkButton = editor.ui.featureComponents.create( 'unlink' );
+				balloonPanel = linkFeature.balloonPanel;
+			} );
+	} );
+
+	afterEach( () => {
+		return editor.destroy();
+	} );
+
+	it( 'should be loaded', () => {
+		expect( linkFeature ).to.instanceOf( Link );
+	} );
+
+	it( 'should load LinkEngine', () => {
+		expect( editor.plugins.get( LinkEngine ) ).to.instanceOf( LinkEngine );
+	} );
+
+	it( 'should register click observer', () => {
+		expect( editor.editing.view.getObserver( ClickObserver ) ).to.instanceOf( ClickObserver );
+	} );
+
+	describe( 'link toolbar button', () => {
+		it( 'should register link feature component', () => {
+			expect( linkButton ).to.instanceOf( Button );
+		} );
+
+		it( 'should bind linkButton#model to link command', () => {
+			const model = linkButton.model;
+			const command = editor.commands.get( 'link' );
+
+			expect( model.isEnabled ).to.be.true;
+
+			command.isEnabled = false;
+			expect( model.isEnabled ).to.be.false;
+		} );
+
+		it( 'should open panel on linkButton#model execute event, when editor is focused', () => {
+			editor.editing.view.isFocused = true;
+
+			linkButton.model.fire( 'execute' );
+
+			expect( linkFeature.balloonPanel.view.isVisible ).to.true;
+		} );
+
+		it( 'should not open panel on linkButton#model execute event, when editor is not focused', () => {
+			linkButton.model.fire( 'execute' );
```

Comment:
This test may be unstable if not set `editor.editing.view.isFocused = false;` first.